LocalFolder/explainer.md

```diff
@@ -0,0 +1,129 @@
+## Proposed Solution 
+
+### Existing behavior
+
+`navigator.storage.getDirectory()` is an existing API that returns a promise to a FileSystemDirectoryHandle representing the root of the origin's OPFS storage space. By default, an origin's OPFS root directory will have no entries.
+
+### New behavior
+
+Within a Microsoft Store PWA context, the directory handle for OPFS root will contain an additional entry that represents the app's LocalFolder location. This entry can be retrieved as a `FileSystemDirectoryHandle` by calling `.entries()` or `.getDirectory(...)` on the OPFS root directory handle.
+
+The LocalFolder directory handle and its contents will have properties different from other OPFS directory handles - they will only have read and delete permissions instead of read and write permissions.
+
+```JS
+    // .getDirectory()
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        let localStorage = await opfsRoot.getDirectoryHandle("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", {create: false}); 
+    }
+```
+
+```JS
+    // .entries()
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        for await (const [key, value] of opfsRoot.entries()) {
+            if (key == `microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt`) {
+                // ...
+            }
+        }
+    }
+```
+
+### Deleting files
+
+LocalStorage contents can be cleared by calling `.removeEntry(...)` from the OPFS root directory handle.
+```JS
+    let opfsRoot = await navigator.storage.getDirectory();
+    if (opfsRoot) {
+        await opfsRoot.removeEntry("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", { recursive: true }); 
+    }
+```
+
+Alternatively, contents in LocalFolder can be deleted by calling `remove()` on file and directory handles directory.
+
+```JS
+    let localStorage = await opfsRoot.getDirectoryHandle("microsoft_store_app_local_folder_APPID.37853FC22B2CE_6rarf9sa4v8jt", {create: false}); 
+
+    for await (const [key, value] of localStorage.entries()) {
+        value.remove();
+    }
+```
+
+### Entry name
+
+The LocalFolder entry within the OPFS root directory can be found under the name `microsoft_store_app_local_folder{PFN}` where `{PFN}` is the Package Family Name of the app.
+
+### Microsoft Store PWA context
+
+To be in a Microsoft Store app context, `navigator.storage.getDirectory()` must be called from:
+* Within a top level document
+* Within a web app window of a Microsoft Store installed PWA
+* From the main thread and not a Web Worker
+
+### Avoiding collisions
+To avoid collisions, the LocalFolder entry under the OPFS root directory is assigned a specific name that contains the Package Family Name. If an entry with the same name already exists under the OPFS root directory, an entry representing the LocalFolder directory will not be made visible until the existing entry is renamed or otherwise removed.
```

Comment:
Would be good to also link to some documentation that explains what a PFN is and how to get it.  I found [this one article](https://learn.microsoft.com/en-us/mem/configmgr/protect/deploy-use/find-a-pfn-for-per-app-vpn) on an easy way to find it, but there may be better ones.  While the article is not specific to VPN's, it's under a VPN topic, so it might be better to find a different one.

Reply:
Added link above in first reference to PFN.